presto-spi/src/main/java/com/facebook/presto/spi/function/QualifiedFunctionName.java

```diff
@@ -38,7 +41,9 @@ public static QualifiedFunctionName of(String dottedName)
     {
         String[] parts = dottedName.split("\\.");
         if (parts.length != 3) {
-            throw new IllegalArgumentException("QualifiedFunctionName should have exactly 3 parts");
+            throw new PrestoException(
+                    INVALID_FUNCTION_ARGUMENT,
+                    format("Non-builtin functions must be reference by three parts: catalog.schema.function_name, found: %s", dottedName));
```

Comment:
@viczhang861 This class cannot possibly know that it is used to describe a non-builtin function. Can you give an example of a query that is failing?

Reply:
@mbasmanova 
This is query and stack trace,  https://gist.github.com/viczhang861/77b4fd0e5cafc0ce1e51b3cd6c57febe
After discussion with Rongrong,  check has been added,  add a comment to remind future caller check argument size before using it.

Reply:
@viczhang861 

> check has been added, add a comment to remind future caller check argument size before using it.

Is there a PR for that? Perhaps, link it here for future reference. Should we close this PR?

Reply:
@mbasmanova  #13561 
@caithagoras  I removed reference to this function as you suggested and only allow test code to use it